app/services/authn_context_resolver.rb

```diff
@@ -54,18 +77,51 @@ def no_identity_proofing_vot
     end
   end
 
+  def acr_results
+    @acr_results ||= decorate_acr_result_with_user_context(
+      acr_result_with_sp_defaults,
+    )
+  end
+
   def acr_result_with_sp_defaults
+    decorate_acr_result_with_sp_defaults(
+      acr_result_without_sp_defaults,
+    )
+  end
+
+  # @param [Vot::Parser::Result] result Parser result to decorate.
+  # @return [Vot::Parser::Result]
+  def decorate_acr_result_with_sp_defaults(result)
     result_with_sp_aal_defaults(
       result_with_sp_ial_defaults(
-        acr_result_without_sp_defaults,
+        result,
       ),
     )
   end
 
+  # @param [Vot::Parser::Result] result Parser result to decorate.
+  # @return [Vot::Parser::Result]
+  def decorate_acr_result_with_user_context(result)
+    return result unless result.biometric_comparison?
+
+    if result.component_values.map(&:name).include?(
+      Saml::Idp::Constants::IAL2_BIO_REQUIRED_AUTHN_CONTEXT_CLASSREF
+    )
+      result.with(biometric_comparison?: true)
+    elsif user&.identity_verified_with_biometric_comparison?
+      result.with(biometric_comparison?: true)
+    elsif user&.identity_verified?
+      result.with(biometric_comparison?: false)
```

Comment:
If `:biometric_comparison` is marked as false, then `:two_pieces_of_fair_evidence` should also be false and vice versa.

This is partially why I started with string substitution, so I could avoid having that type of business logic in the resolver.

Reply:
ah great point! we can add that while we pair (or feel free to add it while applying other changes)

that makes sense, although having had a chat with hooper, it sounds like he intended the resolver to be the place to put that kind of logic, so that we can flexibly apply things depending on user/service provider (maybe other) context when necessary. let's definitely talk about the approach, i ended up changing more than i thought i was going to b/c i ran with hooper's suggestion and i want to make sure you're comfortable with it!

Reply:
Okay, I'll make those changes and see if anything breaks (😅)